### PR TITLE
Fix issue #629 (first attempt)

### PR DESCRIPTION
**Pull request by pwaller**
This is an attempt at fixing #629. I don't have very high confidence in it, but it doesn't seem to obviously break anything.

### CODE REVIEW

**Comment by TheCycoONE** [comment]
This seems pretty much what @jorgenpt was suggesting on irc.

File: CorsixTH/Src/th_gfx_sdl.cpp [hunk 1]
```diff
@@ -397,17 +397,26 @@ bool THRenderTarget::fillRect(uint32_t iColour, int iX, int iY, int iW, int iH)
 void THRenderTarget::getClipRect(THClipRect* pRect) const
 {
     SDL_RenderGetClipRect(m_pRenderer, reinterpret_cast<SDL_Rect*>(pRect));
+    if (pRect->w == 0 || pRect->h == 0) {
```
Comment (jorgenpt) [inline CorsixTH/Src/th_gfx_sdl.cpp]: I would create a temp `pSDLRect`, pass that to `SDL_RenderGetClipRect`, and then check  `SDL_RectEmpty(pSDLRect)`. That way you're consistent with `THRenderTarget::setClipRect`. It should be exactly the same check under the hood, though.

**Comment by jorgenpt** [comment]
Other than that one in-line nitpick, LGTM (Looks Good To Me).

**Comment by pwaller** [comment]
I've made the change whilst from my mobile connection through the github UI, so I haven't even tested if it compiles, but I think it's straightforward enough. PTAL @jorgenpt

**Comment by TheCycoONE** [comment]
Looks alright to me.  All these reinterpret_casts make me a bit nervous but I know they predate this patch.  :+1:

**Comment by TheCycoONE** [comment]
I believe this pr is superseded by #656.  Thanks @pwaller.

**Comment by pwaller** [comment]
Great to see it fixed! Good work @Andy51

**Pull request closed without merging**